src/plugins/data/common/es_query/es_query/filter_matches_index.ts

```diff
@@ -18,5 +18,12 @@ export function filterMatchesIndex(filter: Filter, indexPattern?: IIndexPattern
   if (!filter.meta?.key || !indexPattern) {
     return true;
   }
+
+  // Fixes https://github.com/elastic/kibana/issues/89878
+  // Custom filters may refer multiple fields. Validate the index id only.
```

Comment:
LGTM, 

just what does this mean exactly? `Custom filters may refer multiple fields.`
`filter.meta.key` is empty for custom filter?

Reply:
Yep. It just says `query` because you can't reliably analyze which fields are actually used.

Reply:
Ok, thanks, makes sense 👍